src/controller.rs

```diff
@@ -403,14 +403,104 @@ fn pod_template(config: &WasmCloudHostConfig, _ctx: Arc<Context>) -> PodTemplate
         ..Default::default()
     }];
 
-    let mut volume_mounts = vec![VolumeMount {
+    let mut nats_volume_mounts = vec![VolumeMount {
         name: "nats-config".to_string(),
         mount_path: "/nats/nats.conf".to_string(),
         read_only: Some(true),
         sub_path: Some("nats.conf".to_string()),
         ..Default::default()
     }];
 
+    let mut wasm_volume_mounts: Vec<VolumeMount> = vec![];
+    if let Some(authorities) = &config
+        .spec
+        .certificates
+        .clone()
+        .and_then(|certificates| certificates.authorities)
+    {
+        for authority in authorities.iter() {
```

Comment:
Since we only support a small subset of the [available fields on Volume](https://docs.rs/k8s-openapi/latest/k8s_openapi/api/core/v1/struct.Volume.html#fields), I think it would be good to validate that one of the supported fields was passed in before we attempt to iterate, so making this something like:
```suggestion
    let mut wasm_volume_mounts: Vec<VolumeMount> = vec![];
    if let Some(authorities) = &config
        .spec
        .certificates
        .clone()
        .and_then(|certificates| certificates.authorities)
    {
        for authority in authorities.iter() {
            if authority.secret.is_none() && authority.config_map.is_none() {
                bail!("could not find one of the supported certificate sources")
            }
```